cloud/linode/loadbalancers.go

```diff
@@ -510,18 +511,40 @@ func (l *loadbalancers) getLoadbalancerTags(ctx context.Context, service *v1.Ser
 	return []string{}
 }
 
+func (l *loadbalancers) getLoadbalancerFirewallID(service *v1.Service) (int, error) {
+	fwid, ok := getServiceAnnotation(service, annLinodeCloudFirewallID)
+	if !ok {
+		return -1, nil
+	}
+
+	firewallID, err := strconv.Atoi(fwid)
+	if err != nil {
+		return -1, err
+	}
+	return firewallID, nil
+}
+
 func (l *loadbalancers) createNodeBalancer(ctx context.Context, clusterName string, service *v1.Service, configs []*linodego.NodeBalancerConfigCreateOptions) (lb *linodego.NodeBalancer, err error) {
 	connThrottle := getConnectionThrottle(service)
 
 	label := l.GetLoadBalancerName(ctx, clusterName, service)
 	tags := l.getLoadbalancerTags(ctx, service)
+	firewallID, err := l.getLoadbalancerFirewallID(service)
+	if err != nil {
+		return nil, err
+	}
+
 	createOpts := linodego.NodeBalancerCreateOptions{
 		Label:              &label,
 		Region:             l.zone,
 		ClientConnThrottle: &connThrottle,
 		Configs:            configs,
 		Tags:               tags,
 	}
+
+	if firewallID != -1 {
+		createOpts.FirewallID = firewallID
+	}
 	return l.client.CreateNodeBalancer(ctx, createOpts)
```

Comment:
We're only doing the `Read`/`Create` paths of all the CRUD operations we'd want to support. I understand there are limitations to what linodego supports at this point, but I'm not sure we should be merging in something that doesn't even support modification or deletion (!).

Reply:
@okokes-akamai, It's not a limitation from the linodego. It's a limitation of the linode API itself. It doesn't yet support updating/deleting the firewalls on the nodeBalancer. https://github.com/linode/linode-api-docs/pull/863/files contains changes to linode API. Since upgrades and deletes are not supported yet, this feature could probably be called an alpha/something related in the documentation.